Mathlib/Analysis/SpecialFunctions/ContinuousFunctionalCalculus/Rpow.lean

```diff
@@ -0,0 +1,347 @@
++ Lemmas about how these functions interact with commuting `a` and `b`.
++ Prove the order properties (operator monotonicity and concavity/convexity)
+-/
+
+open scoped NNReal
+
+namespace CFC
+
+section NonUnital
+
+variable {A : Type*} [PartialOrder A] [NonUnitalNormedRing A] [StarRing A] [StarOrderedRing A]
+  [TopologicalRing A] [Module ℝ≥0 A] [SMulCommClass ℝ≥0 A A] [IsScalarTower ℝ≥0 A A]
+  [CompleteSpace A] [NonUnitalContinuousFunctionalCalculus ℝ≥0 (fun (a : A) => 0 ≤ a)]
+
+/- ## `nnrpow` -/
+
+/-- Real powers of operators, based on the non-unital continuous functional calculus. -/
+noncomputable def nnrpow (a : A) (y : ℝ≥0) : A := cfcₙ (fun x : ℝ≥0 => x ^ (y : ℝ)) a
+
+/-- Enable `a ^ y` notation for `CFC.nnrpow`. This is a low-priority instance to make sure it does
+not take priority over other instances when they are available. -/
+noncomputable instance (priority := 50) : Pow A ℝ≥0 where
+  pow a y := nnrpow a y
+
+@[simp]
+lemma nnrpow_eq_pow {a : A} {y : ℝ≥0} : nnrpow a y = a ^ y := rfl
+
+@[simp]
+lemma nnrpow_nonneg {a : A} {x : ℝ≥0} : 0 ≤ a ^ x := cfcₙ_predicate _ a
+
+lemma nnrpow_def {a : A} {y : ℝ≥0} : a ^ y = cfcₙ (fun x : ℝ≥0 => x ^ (y : ℝ)) a := rfl
+
+lemma nnrpow_add {a : A} {x y : ℝ≥0} (hx : 0 < x) (hy : 0 < y) :
+    a ^ (x + y) = a ^ x * a ^ y := by
+  simp only [nnrpow_def]
+  rw [← cfcₙ_mul (fun z : ℝ≥0 => z ^ (x : ℝ)) (fun z : ℝ≥0 => z ^ (y : ℝ)) a]
+  congr
+  ext z
+  have : (x : ℝ) + (y : ℝ) ≠ 0 := by exact_mod_cast (ne_of_gt (add_pos hx hy))
+  simp [NNReal.rpow_add' _ this]
+
+@[simp]
+lemma nnrpow_zero {a : A} : a ^ (0 : ℝ≥0) = 0 := by
+  simp [nnrpow_def, cfcₙ_apply_of_not_map_zero]
+
+lemma nnrpow_one {a : A} (ha : 0 ≤ a := by cfc_tac) : a ^ (1 : ℝ≥0) = a := by
+  simp only [nnrpow_def, NNReal.coe_one, NNReal.rpow_one]
+  change cfcₙ (id : ℝ≥0 → ℝ≥0) a = a
+  rw [cfcₙ_id ℝ≥0 a]
+
+lemma nnrpow_two {a : A} (ha : 0 ≤ a := by cfc_tac) : a ^ (2 : ℝ≥0) = a * a := by
+  simp only [nnrpow_def, NNReal.coe_ofNat, NNReal.rpow_ofNat, pow_two]
+  change cfcₙ (fun z : ℝ≥0 => id z * id z) a = a * a
+  rw [cfcₙ_mul id id a, cfcₙ_id ℝ≥0 a]
+
+lemma nnrpow_three {a : A} (ha : 0 ≤ a := by cfc_tac) : a ^ (3 : ℝ≥0) = a * a * a := by
+  simp only [nnrpow_def, NNReal.coe_ofNat, NNReal.rpow_ofNat, pow_three]
+  change cfcₙ (fun z : ℝ≥0 => id z * (id z * id z)) a = a * a * a
+  rw [cfcₙ_mul id _ a, cfcₙ_mul id _ a, ← mul_assoc, cfcₙ_id ℝ≥0 a]
+
+@[simp]
+lemma zero_nnrpow {x : ℝ≥0} : (0 : A) ^ x = 0 := by simp [nnrpow_def]
+
+@[simp]
+lemma nnrpow_nnrpow [UniqueNonUnitalContinuousFunctionalCalculus ℝ≥0 A]
+    {a : A} {x y : ℝ≥0} : (a ^ x) ^ y = a ^ (x * y) := by
+  by_cases ha : 0 ≤ a
+  case pos =>
+    by_cases hx : 0 < x
+    case neg =>
+      replace hx : x = 0 := eq_of_le_of_not_lt (le_of_not_lt hx) not_lt_zero'
+      simp [hx]
+    case pos =>
+      by_cases hy : 0 < y
+      case neg =>
+        replace hy : y = 0 := eq_of_le_of_not_lt (le_of_not_lt hy) not_lt_zero'
+        simp [hy]
```

Comment:
You can deal with all the `0` cases like this quickly:
```
    obtain (rfl | hx) := eq_zero_or_pos x <;> obtain (rfl | hx) := eq_zero_or_pos y
    all_goals try simp
```
By the way, I knew we had to have that lemma: `eq_zero_or_pos`, and it was found instantly with a quick [Loogle](https://loogle.lean-lang.org/?q=%3Fx+%3D+0+∨+0+<+%3Fx)